lib/chai/core/assertions.js

```diff
@@ -504,10 +499,42 @@ module.exports = function (chai, _) {
       return;
     }
 
-    // Assert inclusion in an array or substring in a string.
+    var isDeep = flag(this, 'deep')
+      , descriptor = isDeep ? 'deep ' : ''
+      , included = false;
+
+    switch (objType) {
+      case 'string':
+        included = obj.indexOf(val) !== -1;
+        break;
+
+      case 'weakmap':
+      case 'weakset':
+        if (isDeep) {
+          var flagMsg = flag(this, 'message')
+            , ssfi = flag(this, 'ssfi');
+          flagMsg = flagMsg ? flagMsg + ': ' : '';
+
+          throw new AssertionError(
+            flagMsg + 'unable to use .deep.include with weak collection',
+            undefined,
+            ssfi
+          );
+        }
+
+        included = obj.has(val);
+        break;
+
+      default:
+        var isEql = isDeep ? _.eql : function(a, b) { return a === b; };
```

Comment:
Well this is a new feature right? So no one would have hit this path before. It would be a small gotcha to say:

> If you pass in (weak)?(map|set) then it'll use samevalue, otherwise ===